telegram/ext/_chatjoinrequesthandler.py

```diff
@@ -43,17 +44,60 @@ async def callback(update: Update, context: CallbackContext)
 
             The return value of the callback is usually ignored except for the special case of
             :class:`telegram.ext.ConversationHandler`.
+        chat_id (:SCT[:obj:`int`], optional): Filters requests to allow only
+            those which are from a specified chat ID.
+
+            .. versionadded:: 20.0
+        username (:SCT[:obj:`str`], optional): Filters requests to allow only
+            those which are from a specified username.
+
+            .. versionadded:: 20.0
         block (:obj:`bool`, optional): Determines whether the return value of the callback should
             be awaited before processing the next handler in
             :meth:`telegram.ext.Application.process_update`. Defaults to :obj:`True`.
 
     Attributes:
         callback (:term:`coroutine function`): The callback function for this handler.
+        _chat_ids (:frozenset(:obj:`int`), optional):
+            Which chat ID(s) to allow through.
+
+            .. versionadded:: 20.0
+        _usernames (:frozenset(:obj:`str`), optional):
+            Which username(s) to allow through.
+
+            .. versionadded:: 20.0
```

Comment:
The whole point behind making attributes private would be to hide it from the users, so no need to document them.